ros2_batch_job/__main__.py

```diff
@@ -356,6 +370,32 @@ def build_and_test(args, job):
     # return 0 if ret_test == 0 and ret_testr == 0 else 1
     return 0
 
+'''
+Adds sanitizer specific mixin as a build arg
+'''
+def update_build_cmd_based_on_sanitizer_args(cmd, args):
+    if args.enable_sanitizer_type == 'address':
+        cmd.extend(['--mixin asan-gcc'])
+    return cmd
+
+'''
+Sets up prerequisites for sanitizer based testing
+'''
+def update_test_cmd_based_on_sanitizer_args(test_cmd, args):
+    '''
+    OSRF memory testing relies on LD_PRELOAD env variable to instrument libc functions like malloc().
+    Sanitizers work in a similar way and hence fail on memory tools tests with the message:
+    "ASan runtime does not come first in initial library list; you should either link runtime to your
+    application or manually preload it with LD_PRELOAD"
+
+    We are overriding LD_PRELOAD with the sanitizer library path before colcon test to resolve this.
+    FIXME(osrf/osrf_testing_tools_cpp#23): Add a cmake argument for disabling memory tool tests and
```

Comment:
Please consider to provide a pull request to add the described CMake option and avoid this.

Reply:
Opened 3 pull requests to add the required cmake option:
https://github.com/ros2/rcutils/pull/135
https://github.com/osrf/osrf_testing_tools_cpp/pull/24
https://github.com/ros2/system_tests/pull/331

We do not need any such override logic in the code anymore